proposals/provenance-logging.md

```diff
@@ -0,0 +1,128 @@
+## Motivation
+Reproducible ML is a [big problem](https://petewarden.com/2018/03/19/the-machine-learning-reproducibility-crisis/). 
+Kubeflow could add a lot of values to users by making it easy for users to track how datasets and models were produced.
+A lot of valuable information is available in K8s resource specifications. Resources like K8s Jobs, TFJobs, ConfigMaps, and Argo workflows
+often contain information (e.g. via command line arguments) about inputs and outputs and the code (docker image) used. There are
+two major problems to using this information to track provenance of models and data today:
+
+1. The information is not durable
+
+	* I should not rely on the K8s APIServer as a persistent, historical data store for resources.
```

Comment:
"... Ideally something immutable with hashing, authentication and version controls would be more appropriate (e.g. Git)."

Reply:
Not sure git is really appropriate in this case but otherwise makes sense and I'll add it to the doc.